app/controllers/spree/stripe_controller.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Spree
+  class StripeController < Spree::BaseController
+    include Core::ControllerHelpers::Order
+
+    def confirm_payment
+      begin
+        if params[:payment_method_id].present?
+          # Create the PaymentIntent
+          intent = stripe.create_intent(
+            (current_order.total * 100).to_i,
+            params[:payment_method_id],
+            currency: current_order.currency,
+            confirmation_method: 'manual',
+            confirm: true,
+            setup_future_usage: 'on_session',
+            metadata: { order_id: current_order.id }
+          )
+        elsif params[:payment_intent_id].present?
+
+          intent = stripe.confirm_intent(params[:payment_intent_id], nil)
+        end
+      rescue Stripe::CardError => e
+        # Display error on client
```

Comment:
Maybe we can remove this line?

Reply:
Sure... comments are from the Stripe documentation example, I should have removed them in the first place 😅